BillTrack.Application/Services/SqsPublisher.cs

```diff
@@ -1,40 +1,47 @@
 using System.Text.Json;
 using Amazon.SQS;
 using Amazon.SQS.Model;
+using BillTrack.Core.Interfaces.Models;
 using BillTrack.Core.Interfaces.Services;
-using Microsoft.Extensions.Caching.Memory;
-using Microsoft.Extensions.Configuration;
 
 namespace BillTrack.Application.Services;
 
 public class SqsPublisher : ISqsPublisher
 {
     private readonly IAmazonSQS _sqs;
-    private readonly IMemoryCache _memoryCache;
-    private readonly IConfiguration _configuration;
+    private IDictionary<string, string> _cachedQueueUrl;
 
-    public SqsPublisher(IAmazonSQS sqs, IMemoryCache memoryCache, IConfiguration configuration)
+
+    public SqsPublisher(IAmazonSQS sqs)
     {
         _sqs = sqs;
-        _memoryCache = memoryCache;
-        _configuration = configuration;
+        _cachedQueueUrl = new Dictionary<string, string>();
     }
 
-    public async Task PublishMessageAsync<T>(T message)
+    public async Task PublishMessageAsync<T>(string queueName, T message) where T : IMessage
     {
-        var queueName = _configuration.GetValue<string>("QueueName");
-
-        var queueUrl = await _memoryCache.GetOrCreateAsync(queueName, async entry =>
+        if (!_cachedQueueUrl.TryGetValue(queueName, out var queueUrl))
```

Comment:
FYI. `out ...` initializers are not obvious to read in most cases. Yes, there are cases when there is no other way than to use this kind of initializer. Also, up to .NET v8 such initialization works much slower than a regular one.